### ISSUE TITLE
wrong metadata album art +1 sugession

### ISSUE THREAD

**Issue by Aneeqasif** [deepjyoti30/ytmdl#184]
# Bug report

## Checklist

- [ x] I have searched for similar bugs in the issues.
- [ x] I will add the bug label to this issue.

## Version Info 

- Ytmdl version: `2021.05.26-1`
- Youtube-dl version: `2021.04.26`

## How Ytmdl was installed?

Aur git version

## Description

mostly the album art of the songs i searched for is wrong mostly covers and things done by other people. i highly suggest to add a flag or something like that, to use YouTube thumbnail as album art of the song there is a python pakage **pythumb** that directly download youtube thumbnails may be u can use this . it will make it one of the best cli music downloaders of all time. waiting for this.
Finally i really wanna appreciate  your efforts on this project its amazing.
Have a good day :heart:

**Comment by deepjyoti30** [comment]
@Aneeqasif Did you select the metadata option yourself? This seems like an issue with the provider of meta. Sometimes the providers do not have proper meta but the song name stays the same, so users select that meta and it ends up with wrong metadata.

Please consider posting a whole log of a song you downloaded. It will help understand the issue better. Just use one song as example (one that you faced the issue with).

Also, regarding adding thumbs from YouTube, you can just use `youube-dl` for that.

**Comment by Aneeqasif** [comment]
```
ytmdl worth it fifth harmony 
 ==> Searching to see if already present in /home/aneeq/Music 
 ==> Searching Youtube for worth it fifth harmony 
 ==> Choose One (One with [M] is verified music) 
 [1] Fifth Harmony - Worth It (Official Video) ft. Kid Ink by Fifth Harmony with dur 3:45
 [2] Fifth Harmony - Worth It (Lyrics) ft. Kid Ink by Loku with dur 4:16
 [3] Worth it - Fifth Harmony ft.Kid Ink / May J Lee Choreography by 1MILLION Dance Studio with dur 8:28
 [4] Fifth Harmony - Worth It (feat. Kid Ink) (Lyrics) by FifthHarmonyHQ with dur 3:46
 [5] Fifth Harmony - 'Worth It' (Summertime Ball 2015) by Capital FM with dur 4:11
 [0] More results
 ==> Enter Choice (1): 1
 ==> Downloading Fifth Harmony - Worth It (Official Video) ft. Kid Ink in 320kbps 
 ==> Saving the files to: /home/aneeq/.cache/ytmdl 
4 MB   | 2 Mb/s || ETA: 3 s                                                                                               |▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓| 100% ==> Downloaded! 
 ==> Getting song data for worth it fifth harmony... 
 ==> Setting data... 
 ==> Choose One  
 [1] Worth It by Helen Dawson
 [2] Worth It by Megan Nicole
 [3] Worth It (feat. Kid Ink) by Fifth Harmony
 [4] Worth It (Made Popular By Fifth Harmony ft. Kid Inc.) [Vocal Version] by Party Tyme Karaoke
 [5] Worth It by Kar Play
 [0] More results
 ==> Enter Choice (1): 0
 [6] Worth It (Originally Performed by Fifth Harmony feat. Kid Ink) by DJ Remix Radio
 [7] Worth It - Tribute to Fifth Harmony and Kid ink by Billboard Masters
 [8] Worth It (Originally Performed by Fifth Harmony & Kid Ink) [Vocal Version] by Singer's Edge Karaoke
 [9] Worth It (In the Style of Fifth Harmony Ft. Kid Ink) [Instrumental Version] by Ya Boy Karaoke
 [10] Worth It (In the Style of Kid Ink and Fifty Harmony) [Instrumental Version] by The Karaoke Studio
 [0] More results
 ==> Enter Choice (1): 3
 ==> Preparing the album cover 
 ==> ================================
  || YEAR: 2015-06-21
  || TITLE: Worth It (feat. Kid Ink)
  || ARTIST: Fifth Harmony
  || ALBUM: Keep Calm & Party
  || GENRE: Pop
  || TRACK NO: 2
  || ALBUM COVER ADDED
 ==> ================================
 ==> Moved to /home/aneeq/Music... 
 ==> Done 

```

i tested 2 or 3 of these options but album art was wrong always.
secondly all the metadata other than album art is good , with youtube-dl i guess your are talking about embed-thumbnails flag but it doesnt actual do it in right way and also other metadata is  missing in that scenario . i  wish there would be away to use album art of youtube thumbnails and all other metadata from ytdml apis. if it is not possible to add it as a feature then kindly guide me a lil bit so that i can acheive this .

**Comment by deepjyoti30** [comment]
@Aneeqasif Thanks for sharing the example.

I tried the song myself and it turns out the album art is different. Mind you, it's not "wrong" from `ytmdl`'s perspective. This is because the covers are provided directly by the sources and `ytmdl` merely just adds that cover. In the case of this song, all the covers are not the correct one, however technically, `ytmdl` is adding the `correct` cover.

Also, this issue should not exist with other songs. You can try out other songs.

Regarding adding album from youtube-dl and meta from `ytmdl`, I am not sure if that is possible in an automated way.

**Comment by Aneeqasif** [comment]
thanks for considering my comment. i hope i would find some way to do that. or it might be added as #185 get in. because there is a flag in youtube-dl called `--add-metadata` that **adds thumbnail as album art** , artist name and album (a lil bit of  metadata) to the song . i guess if you add feature to pass flags to youtube-dl then it might work. Anyways it would be a great feature to add yt thumbnail as album art because ytdml is fetching things from youtube and all things that are available on you tube might not be available, (and sometimes its a cover or some edition of orignal by other artist who add their own album art on the track), on that services  whose apis we are using in ytdml while on the other hand there is perfect album art available of that song on yt. Anyways I'm closing issue hoping it would get resolved as more features got added to ytmdl project. and again a thumbs up  for your effort for providing such a great tool. :+1:

**Comment by deepjyoti30** [comment]
@Aneeqasif I think adding #185 would fix this issue. I will keep it open till that one is implemented. You can then test it and continue the conversation here.

However a thing to note is, after youtube-dl does it's processing, `ytmdl` by default, removes all the previously added meta tags from the song and then adds new ones so there might be need for a little bit of fix but we can discuss that once #185 is added.
